cli/src/common/boto3/ec2.py

```diff
@@ -31,7 +36,15 @@
     @AWSExceptionHandler.handle_client_exception
     def describe_subnets(self, subnet_ids):
         """Return a list of subnets."""
-        return self._paginate_results(self._client.describe_subnets, SubnetIds=subnet_ids)
+        return self._client.describe_subnets(SubnetIds=subnet_ids).get("Subnets", [])
```

Comment:
Why not paginate the result?

Reply:
One reason is that the paginate is useful only when the result contains more than [1000 items](https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html).
This describe subnets method is accepting a list of subnets (that is a limited one) so we don't need to paginate.

The other reason is that `@AWSExceptionHandler.handle_client_exception` is not able to correctly manage the handling the exception iwhen there is a generator from the pagination. Still to be investigated.

Reply:
boto3 is not behaving as neatly as AWS CLI. When I tried to [retrieve free tier instance type](https://github.com/aws/aws-parallelcluster/blob/develop/cli/src/pcluster/utils.py#L1173), I had to use pagination, or I might get empty result

Reply:
Fixed, re-added pagination.